verilog/tools/lint/verilog_lint.cc

```diff
@@ -74,9 +91,24 @@ int main(int argc, char** argv) {
   // All positional arguments are file names.  Exclude program name.
   for (const auto filename :
        verible::make_range(args.begin() + 1, args.end())) {
-    // Copy configuration, so that it can be locally modified per file.
-    const LinterConfiguration config(
-        verilog::LinterConfigurationFromFlags(filename));
+    // local configuration, so that it can be locally modified per file.
+    LinterConfiguration config;
+
+    const verilog::LinterConfigurationFlags flags = {
```

Comment:
So, personally, I'd probably solve this differently: keep the flags wherever they were before, but define the LinterConfigurationFlags struct (or, as suggested LinterConfigurationOptions).
The default constructore of LinterConfigurationOptions would just set the values all from the flags, so that you just would then assign values in main() that you'd like to be overriden.

That way, we keep the encapsulation of flags and keep the use of flags consistent in case we use juse the linter library in another tool. But it also allows for full unit-testing, as all the Options can be set manually.

Reply:
Sure, I can move them back. I moved them here as the official guidelines ([Flags Best Practices](https://abseil.io/docs/cpp/guides/flags)) say:

> * Prefer to define flags only in the file containing the binary’s main() function. Although Abseil flags may be defined anywhere in any namespace, avoid any usage outside of main() as it will otherwise be difficult to resolve conflicts.
> * Prefer to reference flags only from within the file containing the binary’s main() function, for the same reason.

And I thought that's what @fangism meant by cleaning the flags up - these are just guidelines however so I guess we can use a mix of both approaches.

Reply:
Sorry, it was my idea to move them to main(), but they may still remain in the library, but be referenced from main().

Reply:
ah ok. I think it is better for them to be where the struct is defined to set some defaults.

The ABSL guidelines are good, but it is also good to ignore them when it makes more sense :D